### ISSUE TITLE
[django 1.11] Upgrading to bootstrap3

### ISSUE THREAD

**Issue by XaviP** [FreedomCoop/valuenetwork#280]
Now in master branch we have a code compatible dj 1.9 - 1.10 - 1.11, but the problem to upgrade to 1.11 is the external module django-forms-bootstrap.
- In requirements.txt we have django-forms-bootstrap==2.0.3.post1 and it's giving error about it:
`render() must be called with a dict, not a Context`
```
@register.filter
def as_bootstrap(form):
    template = get_template("bootstrap/form.html")
    form = _preprocess_fields(form)

    c = Context({
        "form": form,
    })
    return template.render(c)
```
- If we upgrade to last version, still gives the error, but there's a PR which solves it:
https://github.com/pinax/django-forms-bootstrap/pull/27
- In case we upgrade to last version after the PR is merged, we'll have to re-stylize many forms, as the version of bootstrap is higher than now.

**Comment by bhaugen** [comment]
Alternatively, we could internalize the module and fix it ourselves.

**Comment by bhaugen** [comment]
We could also just fork it and use our fork. I don't know which is simpler, but either way, it's pretty simple stuff. And from what I see over on that repo, nobody is working on it anymore.
https://github.com/pinax/django-forms-bootstrap

Most work done some years ago, last update (a PR) 7 months ago. I suspect this depends on bootstrap 2, which might be why. Might get a response on their slack channel: http://slack.pinaxproject.com/

**Comment by XaviP** [comment]
@escanda said: what if we switched the theming altogether to: 
https://github.com/dyve/django-bootstrap3 ?

A fast search:
```
grep -r '{% load bootstrap_tags %}' ./
grep -r '{{ [a-z_\.]*^\|as_bootstrap }}' ./
```
can be substituted by a bulk change:
```
grep -rl '{% load bootstrap_tags %}' ./ | xargs sed -i s@'{% load bootstrap_tags %}'@'{% load bootstrap3 %}'@g
```
I'm looking for the sed command to substitute `|as_bootstrap`, something like (it doesn't work yet):
```
grep -rl '{{ [a-z_\.]*^\|as_bootstrap }}' ./ | xargs sed -Ei s@'{{ \([a-z_\.]*\)^\|as_bootstrap }}'@'{% bootstrap_form \1 %}'@g
```

**Comment by bhaugen** [comment]
> @escanda said: what if we switched the theming altogether to:
https://github.com/dyve/django-bootstrap3 ?

:+1:

**Comment by XaviP** [comment]
I think it's worth it, only css needs some ajustments, js is already working:
https://github.com/FreedomCoop/valuenetwork/commits/bootstrap3

**Comment by escanda** [comment]
I think as well it would be beneficial for the project to upgrade to bootstrap 3. There's some css work still to do but I think it can be accomplished :)

**Comment by XaviP** [comment]
@bhaugen @bum2 
Dev enviroment instructions to help in upgrading to bootstrap3. If you are in a updated master branch:
```
$ git fetch
$ git checkout bootstrap3
$ source ../env/bin/activate
(env)$ pip uninstall django-forms-bootstrap
(env)$ pip install -r requirements.txt --upgrade --trusted-host dist.pinaxproject.com
```
This installs django-bootstrap3 in your dev enviroment, and now it's ready to push changes to bootstrap3 branch. I think if we improve the main header, footer and menus, and most useful pages, it will be enough to merge the branch into master and deploy it.

**Comment by bhaugen** [comment]
Thanks for the instructions. @XaviP what do you think is the urgency for this set of changes? I'd like to wait until we make sure the process logging features in the work app are working for @skoria and Guy.

**Comment by XaviP** [comment]
Not really urgent, so fix the priorities you think :-)

**Comment by bhaugen** [comment]
Personally, I am just waiting for the process logging features to work for the early users before trying the bootstrap3 branch. Other people might have other priorities, or I might be missing something that I should be working on. But I agree we shd get with bootstrap3.

**Comment by paltman** [comment]
I just merged a fix for the 1.11 issue for django-forms-bootstrap and working on a fresh release to PyPI this morning.

**Comment by paltman** [comment]
3.1.0 is now on PyPI

**Comment by bhaugen** [comment]
Hey @paltman ! Thanks for answering the call. You were part of the original Pinax gang, weren't you?

**Comment by XaviP** [comment]
Changing the title as it's up to the people working on it to make it responsive or not.

**Comment by paltman** [comment]
@bhaugen yes and still am! :)

**Comment by paltman** [comment]
@bhaugen, of course, that is for some value of _original_.  I think you pre-dated me. :)

**Comment by bhaugen** [comment]
>  I think you pre-dated me. :)

Might have, but I never did much work on it, so it hardly counts.

**Comment by XaviP** [comment]
- Internalized pinax_forms_bootstrap. See https://github.com/FreedomCoop/valuenetwork/commit/a7bb4988de01229f9bd3d26e7ab14d35afa6aa7e
- Fixed issue for compatibility on django 1.10
- Deployed upgrade to django 1.10 in testocp

Closing issue. Please, reopen if you feel like adapting to bootstrap 3
